plugins/rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py

```diff
@@ -261,6 +261,52 @@ def _setup_endpoint(self, region_string: str) -> str:
 
         return endpoint
 
+    def get_agents_by_ip(self, ip_address: str) -> List[Dict[str, Any]]:
+        agents = []
+        payload = {
+            "query": "query( $orgId:String! ) { organization(id: $orgId) { assets( first: 10000 ) { pageInfo { hasNextPage endCursor } edges { node { id platform host { vendor version description hostNames { name } primaryAddress { ip mac } uniqueIdentity { source id } attributes { key value } } publicIpAddress agent { agentSemanticVersion agentStatus quarantineState { currentState } } } } } } }",
+            "variables": {"orgId": self.org_key},
+        }
+
+        self.logger.info(f"Getting all agents by IP: {ip_address}...")
+        results_object = self._post_payload(payload)
+
+        has_next_page = (
+            results_object.get("data", {})
+            .get("organization", {})
+            .get("assets", {})
+            .get("pageInfo", {})
+            .get("hasNextPage")
+        )
+        agents.extend(self._get_agents_from_result_object(results_object))
+        self.logger.info("Initial agents received.")
+
+        # See if we have more pages of data, if so get next page and append until we reach the end
+        self.logger.info(f"Extra pages of agents: {has_next_page}")
+        while has_next_page:
+            self.logger.info("Getting next page of agents.")
+            has_next_page, results_object, next_agents = self._get_next_page_of_agents(results_object)
+            agents.extend(next_agents)
+
+        # Filter and return agents
+        self.logger.info("Returning results...")
+        filtered_agents = list(
+            filter(
+                lambda agent_: any(
+                    (
+                        agent_.get("publicIpAddress") == ip_address,
+                        agent_.get("host", {}).get("primaryAddress", {}).get("ip") == ip_address,
+                    )
+                ),
+                agents,
+            ),
+        )
+
+        # Remove 'agent' field name to 'agent_info'
```

Comment:
Suggested reword: 'Remove 'agent' field name from 'agent_info'
@igorski-r7, why is this field removed?

Reply:
Mb, It should be `Rename 'agent'` instead of `Remove` 😄 Updating...